src/Actions/Web3.js

```diff
@@ -10,6 +10,7 @@ export function setWeb3Instance(web3Instance) {
 
 export const SET_RPC_PROVIDER_URL = `${prefix}/SET_RPC_PROVIDER_URL`
 export function setRPCProviderUrl(url) {
+  url = url.replace('0.0.0.0', '127.0.0.1')
```

Comment:
Thanks for your contribution, @wuya666.

Is there someplace else we can put this logic? Putting it here breaks [semantic cohesion](https://en.wikipedia.org/wiki/Cohesion_(computer_science)), as the setter for the client's URL shouldn't need to know anything about what host the server is listening on.

I wish I could make a specific recommendation, but I'm honestly not sure exactly where I'd put it just yet.

Reply:
well, I agree it's an ugly hack, but the listening address 0.0.0.0 is really quite a special address where it means the server is listening on all network addresses, while it's an invalid address for the client to connect to... 

I guess maybe we can split the host into two different variables, where one is for the server to listen on, the other is for the client to connect to, and in the settings, when the host is set to 0.0.0.0, the second host is set to 127.0.0.1, while in all other cases the second host is set the same as the host, but that's still some hard-coded special case handling, and may make the related code parts unneccesarily bloat...

Reply:
well, I agree it's an ugly hack, but the listening address 0.0.0.0 is really quite a special address where it means the server is listening on all network addresses, while it's an invalid address for the client to connect to...

I guess maybe we can split the host into two different variables, where one is for the server to listen on, the other is for the client to connect to, and in the settings, when the host is set to 0.0.0.0, the second host is set to 127.0.0.1, while in all other cases the second host is set the same as the host, but that's still some hard-coded special case handling, and may make the related code parts unneccesarily bloat...

Reply:
In retrospect, I have reversed my opinion on this change. So sorry for the delay, @wuya666, and thanks for your contribution!